#95 Separated users for db in tests, segregated test for connect

kortov opened the pull request:
#95

kortov committed:
#95 Separated users for db in tests, segregated test for connect
- Separated logic for db's admin functional (set testing environment) and ordinary user for working with db
- Segregated test for connecting to the db

ivshapovalov commented on src/test/java/juja/sqlcmd/DatabaseManagerConnectTest.java:
```diff
@@ -0,0 +1,102 @@
+        assertFalse(databaseManager.connect(TEST_DB_NAME, DB_USER_LOGIN, DB_USER_PASSWORD));
+    }
+
+    private static void executeSqlQuery(String sqlQuery) throws SQLException {
```
Extra method

kortov replied:
It's for dropping and creating a table

ivshapovalov commented on src/test/java/juja/sqlcmd/DatabaseManagerConnectTest.java:
```diff
@@ -0,0 +1,102 @@
+        executeSqlQuery("DROP DATABASE IF EXISTS " + TEST_DB_NAME);
+        executeSqlQuery("CREATE DATABASE " + TEST_DB_NAME + " OWNER =" + DB_USER_LOGIN);
+        connection.close();
+        connection = DriverManager.getConnection(JDBC_URL + TEST_DB_NAME, DB_USER_LOGIN, DB_USER_PASSWORD);
```
Extra line

kortov replied:
yep

ivshapovalov commented on src/test/java/juja/sqlcmd/DatabaseManagerConnectTest.java:
```diff
@@ -0,0 +1,102 @@
+        connection = DriverManager.getConnection(JDBC_URL + TEST_DB_NAME, DB_USER_LOGIN, DB_USER_PASSWORD);
+    }
+
+    @AfterClass
```
Extra method

kortov replied:
yep

ivshapovalov commented on src/test/java/juja/sqlcmd/DatabaseManagerTest.java:
```diff
@@ -6,99 +6,56 @@
-        connection = DriverManager.getConnection(
-                JDBC_URL + TEST_DB_NAME, DB_USER_NAME, DB_USER_PASSWORD);
+        connection = DriverManager.getConnection(JDBC_URL + TEST_DB_NAME, DB_ADMIN_LOGIN, DB_ADMIN_PASSWORD);
+        executeSqlQuery("ALTER SCHEMA public OWNER TO " + DB_USER_LOGIN);
```
This line is based on the assumption that postgres create schema public when create new database.
But if not?
I think better way to avoid unwanted behavior is

        executeSqlQuery("DROP SCHEMA IF EXISTS public CASCADE");
        executeSqlQuery("CREATE SCHEMA public AUTHORIZATION " + TEST_DB_USER);

The meaning is the same, sure.

kortov replied:
Actually this line is redundant, but the idea about not existing schema public is interesting

kortov committed:
#95 Fixes acc comments
- fixed initial creating of db schema
- removed extra methods from DbManagerConnectTest

Shtramak commented on src/test/java/juja/sqlcmd/DatabaseManagerConnectTest.java:
```diff
@@ -0,0 +1,95 @@
+    public void setUpStreamsAndDbManager() {
+        databaseManager = new DatabaseManager();
+        originalOut = System.out;
+        originalErr = System.err;
```
For what are these manipulations with PrintStreams?

kortov replied:
There are too many errors and exceptions in the output, I hide them

Shtramak replied:
You could just turn off the Postgres JDBC Logger in connect method  and you won't have any logs in console output

kortov replied:
It doesn’t work, I have tests for jdbc driver

Shtramak commented on src/test/java/juja/sqlcmd/DatabaseManagerConnectTest.java:
```diff
@@ -0,0 +1,95 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class DatabaseManagerConnectTest {
```
If we have DatabaseManagerConnectTest, I think we also should have a class DatabaseManagerConnect

kortov replied:
It's tricky to implement)

AVM82 replied:
Agree with @Shtramak. 
I don't quite understand why you did this, if you can explain it to me please.

kortov replied:
I use connect functionality explicitly in Databasemanager’s tests setUp. To know that connect works normally I moved these test to another class (acc. Ivan’s suggestion)

Shtramak commented on src/test/java/juja/sqlcmd/DatabaseManagerTest.java:
```diff
@@ -6,99 +6,56 @@
 
     @Before
-    public void setUpStreamsAndDbManager() throws SQLException {
+    public void setUpDbManager() throws SQLException {
```
As for me setUp() would be enough, DbManager is redundant because with DatabaseManagerTest you are testing DatabaseManager and therefore all of the methods and test environment are reference to it...

Shtramak commented on src/test/java/juja/sqlcmd/DatabaseManagerTest.java:
```diff
@@ -6,99 +6,56 @@
 
     @After
-    public void cleanUpStreams() throws SQLException {
+    public void closeDbManagerConnection() throws SQLException {
```
closeConnection(), DbManager is redundant (see previous remark)

kortov replied:
There are already closeConnection, and they should be separated

Shtramak replied:
Well, it could be simple close()  or closeResources()... Whatever it's not principal, main thing is that method has understandable name.

Shtramak reviewed (approved):
I realy like your decision to move testing of connection in separate class, but then we also need to extract connection responsibility from DatabaseManager. And that would be a great decision and a peace of <s>headake</s>&#x1F539;headache&#x1F539; also )

kortov committed:
#95 MINOR fixed method's name

kortov committed:
#95 MINOR fixed testmethod's name

AVM82 reviewed (approved):
Nice
